pkg/vectorize/ceil/ceil.go

```diff
@@ -0,0 +1,381 @@
+// Copyright 2021 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ceil
+
+import "math"
+
+/*ceil(1.23) -----> 2.0
+ceil(1.23, 0) ----> 2.0
+ceil(1.23, 1) -----> 1.3
+ceil(12.34, -1) ---- > 20.0*/
+
+var (
+	ceilUint8   func([]uint8, []uint8, int64) []uint8
+	ceilUint16  func([]uint16, []uint16, int64) []uint16
+	ceilUint32  func([]uint32, []uint32, int64) []uint32
+	ceilUint64  func([]uint64, []uint64, int64) []uint64
+	ceilInt8    func([]int8, []int8, int64) []int8
+	ceilInt16   func([]int16, []int16, int64) []int16
+	ceilInt32   func([]int32, []int32, int64) []int32
+	ceilInt64   func([]int64, []int64, int64) []int64
+	ceilFloat32 func([]float32, []float32, int64) []float32
+	ceilFloat64 func([]float64, []float64, int64) []float64
+)
+
+var MaxUint8digits = digits(math.MaxUint8)
+var MaxUint16digits = digits(math.MaxUint16)
+var MaxUint32digits = digits(math.MaxUint32)
+var MaxUint64digits = digits(math.MaxUint64) // 20
+var MaxInt8digits = digits(math.MaxInt8)
+var MaxInt16digits = digits(math.MaxInt16)
+var MaxInt32digits = digits(math.MaxInt32)
+var MaxInt64digits = digits(math.MaxInt64) // 19
+
+func digits(value uint64) int64 {
+	digits := int64(0)
+	for value > 0 {
+		value /= 10
+		digits++
+	}
+	return digits
+}
+
+// scaleTable is a lookup array for digits
+var ScaleTable = [...]uint64{
```

Comment:
perhaps import this ScaleTable and Maxdigits from the floor package would also do the job, and makes code more succinct

Reply:
problem solved